submissions/BlueLamantine/oop/index.js

```diff
@@ -0,0 +1,133 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/BlueLamantine/a-tiny-JS-world
+   Web app: https://bluelamantine.github.io//a-tiny-JS-world/
+   */
+class General {
+  constructor({ name, gender, friends, species }) {
+    this.name = name;
+    this.gender = gender;
+    this.species = species;
+    this.friends = Array.isArray(friends) ? friends : 'no friends';
+  }
+  getGender() {
+    return {
+      male: ['He', 'his'],
+      female: ['She', 'her'],
+    };
+  }
```

Comment:
Effectively this method returns a set of constants. We do not even need a method for this, having instead a static class property as these contants are all the same for all instances.

Alternatively we could make this method static. Then it would be called as `General.getGender()` denoting its general nature as it returns the same value for any instance of the class or its children.

**But let's make it useful as a regular non-static class method. And here is how:**

By the moment this method is called the gender is already assigned. Can you make this method returning only set of pronouns which would depend on current object's gender?
You may even retain the literal object here as a dictionary and just pick a proper pair of pronouns based on `this.gender`.

By the way, rename this method please. It doesn't return any genders at all. What is it that it actully returns?

If you can improve your code as per above, then here is the next challenge:
`super.getGender()[this.gender][0]` looks pretty cryptic.

What if we could tell the method what exactly pronoun case we need?
Just look how the following fragments read:
```
getPronoun('nominative') // returns 'He' or 'She' depending on gender
getPronoun('posessive') // returns 'His' or 'Her' depending on gender
```

Reply:
Challenge accepted and accomplished!